### ISSUE TITLE
Standardization of NEP Number Assignment and Usage

### ISSUE THREAD

**Issue by Jim8y** [neo-project/proposals#173]
Proposing a standardized method for assigning and managing NEO Enhancement Proposal (NEP) numbers to address the logistical challenges involved with the naming and referencing of NEP drafts. Currently, NEP drafts cannot be assigned a permanent NEP number until the proposal is officially adopted, causing confusion and inefficiency. This proposal ensures each draft receives a unique and final NEP number upon submission.

Motivation

The current process for managing NEP numbers is cumbersome and inefficient, causing confusion and delays in the drafting and reviewing phases. Proposers often need to reference NEP numbers in their drafts and related documentation, but without a permanent number, this becomes a moving target until adoption. This proposal seeks to streamline the process by ensuring each draft receives a unique and final NEP number upon submission.

Specification

	1.	NEP Number Assignment:
	•	Upon submission of a draft NEP, a unique NEP number will be assigned immediately.
	•	This number will be considered used, regardless of the draft’s final status (adopted or not).
	2.	Usage of NEP Numbers:
	•	Drafts must use their assigned NEP number in all references within the document.
	•	When submitting a draft as a pull request (PR), the PR title must include the NEP number in the format: Draft NEP XX: [Title].
	•	If a PR includes multiple drafts, the title format should be: Draft NEP XX and Draft NEP YY: [Title].

Rationale

By standardizing the assignment of NEP numbers at the submission stage, this proposal ensures clarity and consistency in how drafts are referenced and managed throughout their lifecycle. This method reduces ambiguity and simplifies the tracking and updating of proposals, making the NEP process more transparent and accessible.

Backwards Compatibility

This proposal does not introduce technical changes to the NEO protocol but modifies the procedural aspects of the NEP process. It is fully backward compatible as it only affects new submissions.

PR: #172

**Comment by shargon** [comment]
I think that is not a problem, we can change the number before merge

**Comment by cschuchardt88** [comment]
Yes, should be the `PR` `number`. Everyone other blockchain uses this numbering system.

**Comment by EdgeDLT** [comment]
> Every other blockchain uses this numbering system

That's not true. Some might use PR numbers during draft stage but every network I have checked so far (including Bitcoin and Ethereum) define sequential numbering as the default. NEP-1 was basically a copy-paste of BIP-1 so it shouldn't come as a surprise.

Which blockchains use PR-based numbering for finalized proposals, and how does it benefit us to suddenly change it after 5 years?

**Comment by Jim8y** [comment]
I don't want to waste time discussing how the number should be. Just give me a solution that does not require me to keep updating the nep number and file name, otherwise, I won't change.

**Comment by cschuchardt88** [comment]
> > Every other blockchain uses this numbering system
> 
> That's not true. Some might use PR numbers during draft stage but every network I have checked so far (including Bitcoin and Ethereum) define sequential numbering as the default. NEP-1 was basically a copy-paste of BIP-1 so it shouldn't come as a surprise.
> 
> Which blockchains use PR-based numbering for finalized proposals, and how does it benefit us to suddenly change it after 5 years?

Ethereum and bitcoin use Issue number

**Comment by EdgeDLT** [comment]
> Ethereum and bitcoin use Issue number

That's the functional result of using sequential numbering and not re-using numbers when a proposal isn't finalized. It's not how BIP-1 or EIP-1 define how a number should be picked.

> Just give me a solution that does not require me to keep updating the nep number, the file name, the related implementations

Each new proposal takes the next number. Don't reuse numbers. Never worry about numbers again.
